lib/rules/no-shadow-route-definition.js

```diff
@@ -182,33 +188,37 @@ function getPropertyByKeyName(objectExpression, keyName) {
 }
 
 function trimRootLevelNestedRoutes(routesPath) {
-  return routesPath.replace(ROOT_PATH_TRIM_REGEX, '/');
+  return routesPath.replace(ROOT_PATH_TRIM_REGEX, URL_CHUNK_SEPARATOR);
 }
 
 function convertPathToGenericForMatching(routePathInfos) {
   return routePathInfos
     .map((routePathInfo) => {
-      const mappedValue = routePathInfo.stringValue.trim();
-      if (routePathInfo.type === SOURCE_PATH_VALUE_TYPE.STATIC) {
-        if (mappedValue.startsWith('/:')) {
-          return '/:generic-param-for-matching';
-        }
-        if (mappedValue.startsWith('/*')) {
-          return '/*generic-wildcard-for-matching';
-        }
-      }
-      if (routePathInfo.type === SOURCE_PATH_VALUE_TYPE.DYNAMIC) {
-        return `/variable:${mappedValue}`;
-      }
-      return mappedValue;
+      const convertedPathForMatchingChunks = routePathInfo.normalizedPath
+        .split(URL_CHUNK_SEPARATOR)
+        .map((pathChunk) => {
+          if (routePathInfo.type === SOURCE_PATH_VALUE_TYPE.STATIC) {
+            if (pathChunk.startsWith(':')) {
+              return ':generic-param-for-matching';
+            }
+            if (pathChunk.startsWith('*')) {
+              return '*generic-wildcard-for-matching';
+            }
+          }
+          if (routePathInfo.type === SOURCE_PATH_VALUE_TYPE.DYNAMIC) {
+            return `variable:${pathChunk}`;
+          }
+          return pathChunk;
+        });
+      return convertedPathForMatchingChunks.join(URL_CHUNK_SEPARATOR);
     })
     .join('');
 }
 
 function convertPathForDisplay(routePathInfos) {
   return routePathInfos
     .map((routePathInfo) => {
-      return routePathInfo.stringValue.trim();
+      return routePathInfo.rawPath;
```

Comment:
For error message we display original path we read from the router.js file for easier mapping which route exactly is conflicting.